pennylane/gradients/finite_difference.py

```diff
@@ -302,7 +302,7 @@ def finite_diff(
             "If this is unintended, please mark trainable parameters in accordance with the "
             "chosen auto differentiation framework, or via the 'tape.trainable_params' property."
         )
-        return [], lambda _: np.zeros([tape.output_dim, len(tape.trainable_params)])
+        return [], lambda _: ()

```

Comment:
Or was this logic moved to the `parameter_shift_hessian` function ?

Reply:
It's what the `qml.grad` and `qml.jacobian` functions do, since they return a tuple with a gradient/jacobian for each trainable QNode parameter. Thus a QNode with no trainable parameters should return an empty tuple. An all zero gradient (which you can see being returned when the output is independent of a trainable parameter) is different than no gradient.

Reply:
Ahh ok makes sense